File: temp_wompi_endpoint_test.mjs
const body = {
  total: 10000,
  customerEmail: 'test@example.com',
  customerPhone: '3123456789',
  reference: 'TEST123456',
  paymentMethod: 'NEQUI',
  customerName: 'Test User',
  products: [{ id: 'prod1', name: 'Test', quantity: 1, unit_price: 10000 }],
};

(async () => {
  try {
    const response = await fetch('https://shelbysebastian-1.onrender.com/api/payments/create-wompi-payment', {
      method: 'POST',
      headers: { 'Content-Type': 'application/json' },
      body: JSON.stringify(body),
    });
    console.log('STATUS', response.status);
    console.log('BODY');
    console.log(await response.text());
  } catch (error) {
    console.error('ERROR', error);
  }
})();

File: temp_wompi_endpoint_test.py
import json
import urllib.request
import urllib.error

url = 'https://shelbysebastian-1.onrender.com/api/payments/create-wompi-payment'
data = {
    'total': 10000,
    'customerEmail': 'test@example.com',
    'customerPhone': '3123456789',
    'reference': 'TEST123456',
    'paymentMethod': 'NEQUI',
    'customerName': 'Test User',
    'products': [{'id': 'prod1', 'name': 'Test', 'quantity': 1, 'unit_price': 10000}],
}
body = json.dumps(data).encode('utf-8')
request = urllib.request.Request(url, data=body, headers={'Content-Type': 'application/json'})

try:
    with urllib.request.urlopen(request, timeout=30) as response:
        print('STATUS', response.status)
        print('RESPONSE_BODY')
        print(response.read().decode('utf-8'))
except urllib.error.HTTPError as error:
    print('HTTP ERROR', error.code)
    print('RESPONSE_BODY')
    print(error.read().decode('utf-8'))
except Exception as error:
    print('ERROR', error)
